validator.go

```diff
@@ -16,41 +17,65 @@ const (
 	ErrNoAccessList     strError = "user role is valid, but denied by default deny on empty access list"
 	ErrAccessNotAllowed strError = "user role is valid, but not allowed by access list"
 	ErrNoParsedClaims   strError = "failed to extract claims"
+	ErrNoTokenFound     strError = "no token found"
 
 	ErrInvalidParsedClaims strError = "failed to extract claims: %s"
 	ErrInvalidSecret       strError = "secret key backend error: %s"
 	ErrInvalid             strError = "%v"
 )
 
-var tokenSources = map[string]bool{
-	"header": true,
-	"cookie": true,
-	"query":  true,
+const tokenSourceHeader = "header"
+const tokenSourceCookie = "cookie"
+const tokenSourceQuery = "query"
+
+var allTokenSources []string
+var tokenSources = map[string]byte{
+	tokenSourceHeader: 0, // the value is the order they are in...
+	tokenSourceCookie: 1,
+	tokenSourceQuery:  2,
+}
+
+func init() { // set the default token_sources up
+	allTokenSources = make([]string, len(tokenSources))
+	for k, v := range tokenSources {
+		allTokenSources[int(v)] = k
+	}
 }
 
+var defaultTokenNames = []string{"access_token", "jwt_access_token"}
+
 // TokenValidator validates tokens in http requests.
 type TokenValidator struct {
 	CommonTokenConfig
-	AuthorizationHeaders map[string]bool
-	Cookies              map[string]bool
-	QueryParameters      map[string]bool
+	AuthorizationHeaders map[string]struct{}
+	Cookies              map[string]struct{}
+	QueryParameters      map[string]struct{}
 	Cache                *TokenCache
 	AccessList           []*AccessListEntry
 	TokenBackends        []TokenBackend
+	TokenSources         []string
 }
 
 // NewTokenValidator returns an instance of TokenValidator
-func NewTokenValidator() *TokenValidator {
-	v := &TokenValidator{}
-	v.AuthorizationHeaders = make(map[string]bool)
-	v.Cookies = make(map[string]bool)
-	v.QueryParameters = make(map[string]bool)
-	v.AuthorizationHeaders["access_token"] = true
-	v.AuthorizationHeaders["jwt_access_token"] = true
-	v.Cookies["access_token"] = true
-	v.Cookies["jwt_access_token"] = true
-	v.QueryParameters["access_token"] = true
-	v.QueryParameters["jwt_access_token"] = true
+func NewTokenValidator(m *AuthProvider) *TokenValidator {
```

Comment:
@njones , as I mentioned earlier, please remove the arguments from this function.